### DISCUSSION TITLE
How to filter on nested relation

### DISCUSSION THREAD

**Discussion by adherb** [prisma/prisma#12052] [Q&A]
I am trying to filter a nested relation by dateTime.

I have the following models:
```
model User {
  id                             Int         @id @default(autoincrement())
  createdAt               DateTime    @default(now())
  email                       String      @unique
  firstName               String      @db.VarChar(50)
  lastName                String      @db.VarChar(50)
  password                String      @db.VarChar(191)
  userWorkouts         UserWorkout[]
}

model UserWorkout {
  id                              Int         @id @default(autoincrement())
  scheduledAt           DateTime?
  completedAt           DateTime?
  isCompleted           Boolean     @default(false)

  user                         User @relation(fields: [userId], references: [id])
  userId                      Int
}
```

I would like to dynamically filter the results by the User 'Id" and the UserWorkout 'scheduledAt' field, bascially I would like to get all results by the month passed in.

I am trying the following query but I am unable to get it to work. Is there something I'm doing wrong or is this not possible yet?

```
export const getUserTrainingDiary = async (req, res) => {
  try {
    const workouts = await prisma.user.findUnique({
      where: {
        id: userId,
         userWorkouts: {
           every: {
               scheduledAt: {
                 gte: firstDayOfMonth,
                 lt: lastDayOfMonth,
             },
           },
         },
      },
      select: {
        id: true,
        firstName: true,
        lastName: true,
        userWorkouts: {
          select: {
            id: true,
            scheduledAt: true,
            isCompleted: true,
          },
        },
      },
    });
    res.json(workouts);
  } catch (err) {
    console.log(err);
  }
};
```

**Comment by psibean** [comment]
Hey, in your code example I can't see where you've defined the `firstDayOfMonth` and `lastDayOfMonth` variables - are these DateTime objects? I assume they are, in which case, as far as I can tell this looks like it should work.

So what happens? Does it throw an error or is the query just not returning the results you're expecting? Sample?

**Reply by vladi-stevanovic** [reply]
👋 Hello @adherb  I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
